src/auth/src/credentials/idtoken.rs

```diff
@@ -27,6 +66,207 @@ use std::sync::Arc;
 use std::time::{Duration, SystemTime, UNIX_EPOCH};
 use tokio::time::Instant;
 
+pub mod mds {
+    //! Credentials for authenticating with [ID tokens] using [Metadata Service].
+    //!
+    //! Google Cloud environments such as [Google Compute Engine (GCE)][gce-link],
+    //! [Google Kubernetes Engine (GKE)][gke-link], or [Cloud Run] provide a metadata service.
+    //! This is a local service to the VM (or pod) which (as the name implies) provides
+    //! metadata information about the VM. The service also provides access
+    //! tokens associated with the [default service account] for the corresponding
+    //! VM. This module provides a builder for `IDTokenCredentials`
+    //! from such metadata service.
+    //!
+    //! The default host name of the metadata service is `metadata.google.internal`.
+    //! If you would like to use a different hostname, you can set it using the
+    //! `GCE_METADATA_HOST` environment variable.
+    //!
+    //! These credentials are commonly used for [service to service authentication].
+    //! For example, when services are hosted in Cloud Run or mediated by Identity-Aware Proxy (IAP).
+    //! ID tokens are only used to verify the identity of a principal. Google Cloud APIs do not use ID tokens
+    //! for authorization, and therefore cannot be used to access Google Cloud APIs.
```

Comment:
I think this paragraph appears with slightly different text on each submodule, please clean them up so they all say the same thing.